src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Serialization/JsonSerializerExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Serialization;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Serialization;
+
+internal static class JsonSerializerExtensions
+{
+    public static void RegisterRazorConverters(this JsonSerializer serializer)
+    {
+        if (serializer is null)
+        {
+            throw new ArgumentNullException(nameof(serializer));
+        }
+
+        // In all of the below we add our converters to both the serializer settings and the actual
+        // JsonSerializer. The reasoning behind this choice is that OmniSharp framework is not consistent
+        // in using one over the other so we want to protect ourselves.
```

Comment:
Out of date.. so do we still need to do this twice?

Reply:
Good catch, we no longer duplicate this.